coins/src/adapters/moneyMarkets/morpho.ts

```diff
@@ -0,0 +1,277 @@
+import { runInPromisePool } from "@defillama/sdk/build/generalUtil";
+import { getApi } from "../utils/sdk";
+import { request } from "graphql-request";
+import { getCurrentUnixTimestamp } from "../../utils/date";
+import { getBlock } from "@defillama/sdk/build/util/blocks";
+import fetch from "node-fetch";
+import { Write } from "../utils/dbInterfaces";
+import { addToDBWritesList } from "../utils/database";
+import { getTokenInfoMap } from "../utils/erc20";
+
+type VaultDatas = {
+  [vault: string]: {
+    totalAssets: number;
+    positions: any[];
+    markets: string[];
+  };
+};
+
+const listaConfig: { [chain: string]: { vault: string; vaultInfo: string } } = {
+  bsc: {
+    vault: "0x8F73b65B4caAf64FBA2aF91cC5D4a2A1318E5D8C",
+    vaultInfo:
+      "https://api.lista.org/api/moolah/vault/list?page=1&pageSize=1000",
+  },
+  ethereum: {
+    vault: "0xf820fB4680712CD7263a0D3D024D5b5aEA82Fd70",
+    vaultInfo:
+      "https://api.lista.org/api/moolah/vault/list?page=1&pageSize=1000&chain=ethereum",
+  },
+};
+
+async function fetchMorphoVaultAddresses(chainId: string) {
+  let assets: { [vault: string]: string } = {};
+  let skip = 0;
+  let length = 1000;
+
+  while (length == 1000) {
+    const query = `
+        query {
+            vaults (first: ${length}, skip: ${skip}, orderBy: Address, where:  {
+                chainId_in: [${chainId}]
+            }) {
+                items {
+                    asset {
+                        address
+                    }
+                    address
+                }
+            }}`;
+
+    const res: any = await request("https://api.morpho.org/graphql", query);
+    res.vaults.items.forEach((item: any) => {
+      assets[item.address.toLowerCase()] = item.asset.address.toLowerCase();
+    });
+    length = res.vaults.items.length;
+    skip += length;
+  }
+
+  return assets;
+}
+
+async function morpho(
+  timestamp: number = 0,
+  vaultAssets: { [vault: string]: string } = {},
+  api: any,
+  target: string
+) {
+  const threeDaysAgo =
+    (timestamp == 0 ? getCurrentUnixTimestamp() : timestamp) - 3 * 24 * 60 * 60;
+  const threeDaysAgoBlock = await getBlock(api.chain, threeDaysAgo);
```

Comment:
instead of getBlock, create another api, say `const apiThreeDaysAgo = new ChainApi(...)`